crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -1,12 +1,188 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::Arguments;
+use crate::comments::{dangling_node_comments, leading_node_comments};
+use crate::context::NodeLevel;
+use crate::other::arg::ArgumentKind;
+use crate::prelude::*;
+use crate::trivia::{first_non_trivia_token, SimpleTokenizer, Token, TokenKind};
+use crate::FormatNodeRule;
+use ruff_formatter::{format_args, write, FormatError};
+use ruff_python_ast::node::{AnyNodeRef, AstNode};
+use rustpython_parser::ast::{Arg, Arguments, Expr, Ranged};
+use std::usize;
 
 #[derive(Default)]
 pub struct FormatArguments;
 
 impl FormatNodeRule<Arguments> for FormatArguments {
     fn fmt_fields(&self, item: &Arguments, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let Arguments {
+            range: _,
+            posonlyargs,
+            args,
+            defaults,
+            vararg,
+            kwonlyargs,
+            kw_defaults,
+            kwarg,
+        } = item;
+
+        let saved_level = f.context().node_level();
+        f.context_mut().set_node_level(NodeLevel::Expression);
+
+        let format_inner = format_with(|f: &mut PyFormatter| {
+            let separator = format_with(|f| write!(f, [text(","), soft_line_break_or_space()]));
+            let mut joiner = f.join_with(separator);
+            let mut last_node: Option<AnyNodeRef> = None;
+
+            let mut defaults = std::iter::repeat(None)
```

Comment:
strange decision on the grammar or parser side to emit the default separate from the argument

Reply:
Haha yeah. I got it wrong so many times. The most recent version of `RustPython` now tracks the `default` on the argument but I haven't had time to upgrade yet.